### ISSUE TITLE
Please add support for tiling the pcb x,y-times

### ISSUE THREAD

**Issue by koppi** [pcb2gcode/pcb2gcode#16]
It would be super handy to mill more than one pcb during the fabrication process.

Please add support for tiling the pcb x and y times in x and y direction.

**Comment by Corna** [comment]
Unfortunately this is not very easy to implement due to the huge number of differences between the cnc controllers... While it is quite easy with LinuxCNC (the entire project can be a subroutine that can be called multiple times with different coordinate system values), it is very hard with grbl, and the only possible solution is to concatenate the same code multiple times
But I realise that this is an useful feature, I'll try to implement it

**Comment by koppi** [comment]
The solution with concatenation sounds ok to me. I was starting to hack two new options ```--tile-x``` and ```tile-y``` into ```options.cpp```, then hacked into ```ngc_exporter.cpp``` the tiling stuff, but I do not know howto add this feature into ```autoleveler.cpp```. Maybe you know better howto write code for all of this, I'll be happy to test it out then.

**Comment by Corna** [comment]
You have to use subroutines if the user specify a software that supports them (LinuxCNC, Mach3 or Mach4), you can't just force the concatenation for every controller
It is not an easy enhancement, I'll do it in the following

**Comment by koppi** [comment]
I'm using OpenSCAM to preview the ngc files. I'm using LinuxCNC to mill these files. I noticed, that OpenSCAM does not support subroutines, so usually I generate two versions of the ngc files: a version with probing and a version without probing. For the tiling feature, it would be ok for me, if the ngc files do not use additional subroutines but rather get bigger in size.

**Comment by Corna** [comment]
Code concatenation is a really bad choice and I don't want to use it unless really necessary
Why don't you use linuxcnc-sim as a simulator?

**Comment by koppi** [comment]
Did you already start implementing this feature request?

I just made a video showing the milling process: https://www.youtube.com/watch?v=2nGcliWefek

**Comment by Corna** [comment]
Unfortunately I'm very busy these days, I'll resume my work on pcb2gcode in the next months

**Comment by corna** [comment]
Hi
I've almost done it, are you available for testing?

**Comment by koppi** [comment]
:+1:

**Comment by corna** [comment]
Great! I just have to finish the drill part, then I'll push it in a separate branch
Do you use only LinuxCNC?

**Comment by corna** [comment]
I'm not in a hurry, and I'll complete it in a few days

**Comment by corna** [comment]
Here it is
https://github.com/corna/pcb2gcode/tree/travis_containers
I haven't updated the man yet; to use it just specify __--tile-x=[number of columns]__ and __--tile-y=[number of rows]__. Please test it both with __--software=LinuxCNC__ and __--software=Custom__.
Thanks

**Comment by koppi** [comment]
I'm stuck at:
```
configure: error: cannot find the flags to link with Boost program_options
```
Ubuntu 15.10 here, do you have any hints? Here's the full log (created with ```script pcb2gcode.log```):

```
$ git clone https://github.com/corna/pcb2gcode
Cloning into 'pcb2gcode'...
remote: Counting objects: 1230, done.[K
remote: Compressing objects:   5% (1/20)   [K
remote: Compressing objects:  10% (2/20)   [K
remote: Compressing objects:  15% (3/20)   [K
remote: Compressing objects:  20% (4/20)   [K
remote: Compressing objects:  25% (5/20)   [K
remote: Compressing objects:  30% (6/20)   [K
remote: Compressing objects:  35% (7/20)   [K
remote: Compressing objects:  40% (8/20)   [K
remote: Compressing objects:  45% (9/20)   [K
remote: Compressing objects:  50% (10/20)   [K
remote: Compressing objects:  55% (11/20)   [K
remote: Compressing objects:  60% (12/20)   [K
remote: Compressing objects:  65% (13/20)   [K
remote: Compressing objects:  70% (14/20)   [K
remote: Compressing objects:  75% (15/20)   [K
remote: Compressing objects:  80% (16/20)   [K
remote: Compressing objects:  85% (17/20)   [K
remote: Compressing objects:  90% (18/20)   [K
remote: Compressing objects:  95% (19/20)   [K
remote: Compressing objects: 100% (20/20)   [K
remote: Compressing objects: 100% (20/20), done.[K
Receiving objects:   0% (1/1230)   
Receiving objects:   1% (13/1230)   
Receiving objects:   2% (25/1230)   
Receiving objects:   3% (37/1230)   
Receiving objects:   4% (50/1230)   
Receiving objects:   5% (62/1230)   
Receiving objects:   6% (74/1230)   
Receiving objects:   7% (87/1230)   
Receiving objects:   8% (99/1230)   
Receiving objects:   9% (111/1230)   
Receiving objects:  10% (123/1230)   
Receiving objects:  11% (136/1230)   
Receiving objects:  12% (148/1230)   
Receiving objects:  13% (160/1230)   
Receiving objects:  14% (173/1230)   
Receiving objects:  15% (185/1230)   
Receiving objects:  16% (197/1230)   
Receiving objects:  17% (210/1230)   
Receiving objects:  18% (222/1230)   
Receiving objects:  19% (234/1230)   
Receiving objects:  20% (246/1230)   
Receiving objects:  21% (259/1230)   
Receiving objects:  22% (271/1230)   
Receiving objects:  23% (283/1230)   
Receiving objects:  24% (296/1230)   
Receiving objects:  25% (308/1230)   
Receiving objects:  26% (320/1230)   
Receiving objects:  27% (333/1230)   
Receiving objects:  28% (345/1230)   
Receiving objects:  29% (357/1230)   
Receiving objects:  30% (369/1230)   
Receiving objects:  31% (382/1230)   
Receiving objects:  32% (394/1230)   
Receiving objects:  33% (406/1230)   
Receiving objects:  34% (419/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  35% (431/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  36% (443/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  37% (456/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  38% (468/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  39% (480/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  40% (492/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  41% (505/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  42% (517/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  43% (529/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  44% (542/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  45% (554/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  46% (566/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  47% (579/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  48% (591/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  49% (603/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  50% (615/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  51% (628/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  52% (640/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  53% (652/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  54% (665/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  55% (677/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  56% (689/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  57% (702/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  58% (714/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  59% (726/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  60% (738/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  61% (751/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  62% (763/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  63% (775/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  64% (788/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  65% (800/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  66% (812/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  67% (825/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  68% (837/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  69% (849/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  70% (861/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  71% (874/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  72% (886/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  73% (898/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  74% (911/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  75% (923/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  76% (935/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  77% (948/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  78% (960/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  79% (972/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  80% (984/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  81% (997/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  82% (1009/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  83% (1021/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  84% (1034/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  85% (1046/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  86% (1058/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  87% (1071/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  88% (1083/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  89% (1095/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  90% (1107/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  91% (1120/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  92% (1132/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  93% (1144/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  94% (1157/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  95% (1169/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  96% (1181/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  97% (1194/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  98% (1206/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects:  99% (1218/1230), 436.01 KiB | 850.00 KiB/s   
remote: Total 1230 (delta 3), reused 0 (delta 0), pack-reused 1210[K
Receiving objects: 100% (1230/1230), 436.01 KiB | 850.00 KiB/s   
Receiving objects: 100% (1230/1230), 923.95 KiB | 850.00 KiB/s, done.
Resolving deltas:   0% (0/811)   
Resolving deltas:   1% (10/811)   
Resolving deltas:   4% (36/811)   
Resolving deltas:   8% (67/811)   
Resolving deltas:   9% (78/811)   
Resolving deltas:  12% (101/811)   
Resolving deltas:  16% (130/811)   
Resolving deltas:  27% (222/811)   
Resolving deltas:  31% (254/811)   
Resolving deltas:  33% (273/811)   
Resolving deltas:  35% (289/811)   
Resolving deltas:  37% (305/811)   
Resolving deltas:  41% (335/811)   
Resolving deltas:  43% (349/811)   
Resolving deltas:  44% (362/811)   
Resolving deltas:  45% (368/811)   
Resolving deltas:  47% (389/811)   
Resolving deltas:  48% (392/811)   
Resolving deltas:  49% (401/811)   
Resolving deltas:  54% (439/811)   
Resolving deltas:  56% (456/811)   
Resolving deltas:  58% (474/811)   
Resolving deltas:  59% (481/811)   
Resolving deltas:  75% (616/811)   
Resolving deltas:  76% (620/811)   
Resolving deltas:  77% (625/811)   
Resolving deltas:  78% (639/811)   
Resolving deltas:  79% (643/811)   
Resolving deltas:  81% (658/811)   
Resolving deltas:  82% (666/811)   
Resolving deltas:  85% (695/811)   
Resolving deltas:  86% (698/811)   
Resolving deltas:  87% (707/811)   
Resolving deltas:  92% (750/811)   
Resolving deltas:  95% (775/811)   
Resolving deltas:  96% (779/811)   
Resolving deltas:  97% (790/811)   
Resolving deltas:  98% (800/811)   
Resolving deltas:  99% (805/811)   
Resolving deltas: 100% (811/811)   
Resolving deltas: 100% (811/811), done.
Checking connectivity... done.
[01;32mkoppi@x220[00m:[01;34m~/tmp[00m$ cd pcb2gcode/
[01;32mkoppi@x220[00m:[01;34m~/tmp/pcb2gcode[00m $ git checkout origin/travis_containers 
Note: checking out 'origin/travis_containers'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 331162a... Tiling WIP
[01;32mkoppi@x220[00m:[01;34m~/tmp/pcb2gcode[00m$ autoreconf -i
libtoolize: putting macros in AC_CONFIG_MACRO_DIR, `m4'.
libtoolize: copying file `m4/libtool.m4'
libtoolize: copying file `m4/ltoptions.m4'
libtoolize: copying file `m4/ltsugar.m4'
libtoolize: copying file `m4/ltversion.m4'
libtoolize: copying file `m4/lt~obsolete.m4'
configure.ac:17: installing 'm4/compile'
[01;32mkoppi@x220[00m:[01;34m~/tmp/pcb2gcode[00m$ ./configure 
checking whether make supports nested variables... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
/home/koppi/tmp/pcb2gcode/m4/missing: Unknown `--is-lightweight' option
Try `/home/koppi/tmp/pcb2gcode/m4/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for g++... g++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking for style of include used by make... GNU
checking dependency style of g++... gcc3
checking for gcc... gcc
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... yes
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for Boost headers version >= 1.47.0... yes
checking for Boost's header version... 1_58
checking for the toolset name used by Boost for g++... configure: WARNING: could not figure out which toolset name to use for g++

checking boost/program_options.hpp usability... yes
checking boost/program_options.hpp presence... yes
checking for boost/program_options.hpp... yes
checking for the Boost program_options library... no
configure: error: cannot find the flags to link with Boost program_options
[01;32mkoppi@x220[00m:[01;34m~/tmp/pcb2gcode[00m$ exit

Script done on Do 19 Nov 2015 16:33:12 CET
```

**Comment by koppi** [comment]
Update: I just fixed this by installing the package ```libboost-program-options1.58-dev```.

**Comment by koppi** [comment]
I added to the Makefile at https://github.com/koppi/arduino-shield :

```
--tile-x=1 --tile-y=2
```
and got a segmentation fault:
```
koppi@x220:~/Dropbox/cnc/arduino-shield$ make
#./drl-optimize.py arduino-shield.drl > arduino-shield.opt.drl
# without probing
pcb2gcode --tile-x=1 --tile-y=2 --basename=ngc/arduino-shield --drill=arduino-shield.drl --front=arduino-shield-F_Cu.gbr --back=arduino-shield-B_Cu.gbr --outline=arduino-shield-Cmts_User.gbr --output-dir=. --postamble=ngc/postamble.ngc --preamble=ngc/preamble.ngc --dpi=1000 --metric=true --metricoutput=true --mirror-absolute=false --optimise=true --zchange=40.0000 --zero-start=true --zsafe=0.5000 --extra-passes=0 --mill-feed=600 --mill-speed=10000 --offset=1.0 --zwork=-0.075 --drill-feed=1000 --drill-front=false --drill-speed=20000 --milldrill=false --nog81=false --onedrill=true --zdrill=-2.0 --cut-front=true --cut-feed=600 --cut-infeed=10.0000 --cut-speed=10000 --cutter-diameter=2.0000 --fill-outline=true --outline-width=1 --zcut=-1.6500 --al-back=false --al-front=false --al-probefeed=100 --al-x=5.0000 --al-y=5.0000 --software=LinuxCNC
Importing preamble... DONE
Importing postamble... DONE
Importing front side... DONE.
Importing back side... DONE.
Importing outline... DONE.
Exporting back... 
Warning: pcb2gcode hasn't been able to fulfill all clearance requirements and tried a best effort approach instead. You may want to check the g-code output and possibly use a smaller milling width.
Makefile:2: recipe for target 'all' failed
make: *** [all] Segmentation fault
```
Howto debug this?

**Comment by koppi** [comment]
Yay – I just played  with the above Makefile, changed it to ```--tile-x=2 --tile-y=2``` [...] ```--software=Custom```, this produces quite promising results:

* ```openscam ngc/arduino-shield_outline.ngc```:
![screenshot from 2015-11-19 16-58-44](https://cloud.githubusercontent.com/assets/9458631/11275656/fd6eac38-8ede-11e5-8bc6-a27e95efb93e.png)
* ```openscam ngc/arduino-shield_front.ngc```:
![screenshot from 2015-11-19 16-59-17](https://cloud.githubusercontent.com/assets/9458631/11275657/fee0fe86-8ede-11e5-9e8f-28f026b52890.png)
* ```openscam ngc/arduino-shield_drill.ngc``` – the drill file part gets somehow not correct:
![screenshot from 2015-11-19 16-59-44](https://cloud.githubusercontent.com/assets/9458631/11275660/030ab5f6-8edf-11e5-99d1-45fc64b4d938.png)

**Comment by corna** [comment]
The drill file is ok too. The weird path is caused by the optimisation: if, for example, you have two bit sizes, pcb2gcode drills the holes for the first bit on all the boards, then it changes bit and it does the remaining ones. With this workflow you have to change each bit only once.
Just to be sure, can you post the drill.ngc file?

I don't understand the segfault error though, I'll investigate.

**Comment by koppi** [comment]
Here's the ```arduino-shield_drill.ngc```: http://paste.ubuntu.com/13350539/ When I open the file in OpenSCAM I can see, that the boards overlap vertically and there's too much space horizontally.

**Comment by corna** [comment]
You're right, I didn't noticed the overlap; I'll investigate

**Comment by corna** [comment]
It should work now, I've fixed the segfault and the overlap bugs in commits corna/pcb2gcode@250d7b9dceb96e90a18909557c4a4ade5f202c4f and corna/pcb2gcode@255378f1ecb9110fd9c5e4feeac0f42774879c7d.
Can you test it? Thanks.

**Comment by koppi** [comment]
Thank's for the quick fixes.

One small thing I noticed, which can be seen in the screenshots above: the origin of the ```arduino-shield_drill.ngc``` file is different to the origin of the ```arduino-shield_front.ngc``` and ```arduino-shield_outline.ngc```.

**Comment by corna** [comment]
That's because you've used ```--drill-front=false```, try with ```--drill-front=true```
(Side note: in you're repo there's no license file; did you do that on purpose?)

**Comment by koppi** [comment]
Thank's for the hint with the drill file, that worked.

For the license file part, do you have any suggestions? I was using the KiCAD library at https://github.com/freetronics/freetronics_kicad_library – not being a lawyer, can you recommend what to do in this case?

**Comment by corna** [comment]
[freetronics_kicad_library](https://github.com/freetronics/freetronics_kicad_library) is released under under CC-BY-NC (see [here](https://github.com/freetronics/freetronics_kicad_library/blob/master/LICENSE)), so you must release your project under either the same license (C BY-NC-SA) or under a [compatible license](https://creativecommons.org/compatiblelicenses). Unfortunately the author have added the NonCommercial clause, therefore the GPL is probably not compatible.
TL;DR: use the same license, [CC BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/)
